### DISCUSSION TITLE
Facing issue with OPA watch flag not picking up the changes.

### DISCUSSION THREAD

**Discussion by rest17** [open-policy-agent/community#159] [OPA and Rego] [answered]
Setup Details - Running opa in docker with bundle and watch command -run --server --log-level debug --log-format json-pretty --watch --bundle with the mounted directory. 
This is intermittent issue  where we see that any change to policy rego files under the mountted directory , the file chnage event is not getting fired and opa not reloading the policy changes. Restarting docker solves the problem. There is no definite behavior . 

Is there any bug or fixes available around this issue.

**Comment by anderseknert** [comment, marked as answer]
Hi @rest17 👋 

Yeah, at least at this point in time, I think the watch functionality is mainly good for development/debugging. I've seen a few reports like this in the past, and I know for a fact file watchers don't work at all for some clustered/distributed file stores. These problems are really hard to track down and fix, and since the feature has mostly been used for development, I don't think there's been a lot of effort poured into improvements here. I wish I had a better answer, but I'd seriously consider just setting up an nginx (or whatever file server you prefer) to server bundles from that for anything more than development or testing.

**Reply by rest17** [reply]
Thanks Anders ,

What would be your suggestion here. Should we use bundle server API to propagate polices changes to opa.

**Reply by anderseknert** [reply]
Yes, that would be my suggestion.
